lmp/dataset.py

```diff
@@ -1,29 +1,39 @@
 import torch.utils.data
 import torch.nn.utils.rnn
+from typing import List, Tuple, Union
+import lmp.tokenizer
+import lmp.config
+
 
 class BaseDataset(torch.utils.data.Dataset):
-    def __init__(self, **kwargs):
+    def __init__(self, text_list: List[str],
+                 config: lmp.config.BaseConfig,
+                 tokenizer: Union[lmp.tokenizer.BaseTokenizer, lmp.tokenizer.BaseTokenizerDict],
+                 uncase: bool = False):
         super(BaseDataset, self).__init__()
 
-        self.text_list = kwargs.pop('text_list', [])
-        self.config = kwargs.pop('config', None)
-        self.tokenizer = kwargs.pop('tokenizer', None)
-
-        self.tokenizer.build_dict(self.text_list, self.config.min_count)
+        self.text_list = text_list
+        self.config = config
+        self.tokenizer = tokenizer
+        self.uncase = uncase  # 是否把大小寫視為不同字, default: False
```

Comment:
when value of variable is `bool` type, named the variable using prefix `is_`.
for example: `is_uncased`